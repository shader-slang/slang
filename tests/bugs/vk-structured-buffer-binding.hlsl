//TEST:SIMPLE: -profile ps_4_0 -entry main -target glsl
///////TEST:REFLECTION:-profile ps_4_0 -target spirv

[[vk::binding(3, 4)]]
RWStructuredBuffer<uint> gDoneGroups : register(u3);

float4 main(
    float3 uv : UV)
    : SV_Target
{
    return gDoneGroups[int(uv.z)]; 
}