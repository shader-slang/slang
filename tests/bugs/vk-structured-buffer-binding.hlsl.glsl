// vk-structured-buffer-binding.hlsl.glsl
//TEST_IGNORE_FILE:

#version 450

#define gDoneGroups gDoneGroups_0
#define uv _S3
#define SV_Target _S2

layout(std430, binding = 3, set = 4)
buffer _S1
{
    uint _data[];
} gDoneGroups;

layout(location = 0)
out vec4 SV_Target;

layout(location = 0)
in vec3 uv;

void main()
{
    SV_Target = vec4(gDoneGroups._data[uint(int(uv.z))]);
    return;
}
