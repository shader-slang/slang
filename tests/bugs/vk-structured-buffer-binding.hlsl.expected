result code = 0
standard error = {
}
standard output = {
#version 450
layout(row_major) uniform;
layout(row_major) buffer;

#line 5 0
layout(std430, binding = 3, set = 4) buffer _S1 {
    uint gDoneGroups_0[];
};

#line 7
layout(location = 0)
out vec4 _S2;


#line 7
layout(location = 0)
in vec3 _S3;


#line 7
void main()
{
    _S2 = vec4(gDoneGroups_0[uint(int(_S3.z))]);
    return;
}

}
